### PR TITLE
ci: add build image

### PR DESCRIPTION
**Pull request by AlinsRan**
Close: #420 

remove `apisix-all-in-one` ci: 
It is meaningless to use the docker hub online image to run CI. It cannot recognize the error of PR. The CI should be run using a local built image.

### CODE REVIEW

File: .github/workflows/apisix-docker-example-test.yaml [hunk 1]
```diff
@@ -9,17 +9,52 @@ on:
       - 'docs/**'
       - '**/*.md'
   pull_request:
-    paths:
-      - 'example/*'
+    branches:
+      - master
```
Comment (tao12345666333) [inline .github/workflows/apisix-docker-example-test.yaml:13]: add a branch for 2.15?

Comment (soulbird) [reply .github/workflows/apisix-docker-example-test.yaml:13]: Why do we need to add branch 2.15?

Comment (AlinsRan) [reply .github/workflows/apisix-docker-example-test.yaml:13]: I think it's because 2.15.x is the LTS version?

Comment (soulbird) [reply .github/workflows/apisix-docker-example-test.yaml:13]: but： 
```
env:
  APISIX_VERSION: "3.1.0"
```

Comment (AlinsRan) [reply .github/workflows/apisix-docker-example-test.yaml:13]: Then APISIX_VERSION will need to be applied to the release/2.15. x branch.

Comment (AlinsRan) [reply .github/workflows/apisix-docker-example-test.yaml:13]: > but：
> 
> ```
> env:
>   APISIX_VERSION: "3.1.0"
> ```

@tao12345666333 PTAL.

Comment (tao12345666333) [reply .github/workflows/apisix-docker-example-test.yaml:13]: You can read variables from GitHub ref

Comment (AlinsRan) [reply .github/workflows/apisix-docker-example-test.yaml:13]: Done

**Review by tao12345666333 — APPROVED** [review]

File: .github/workflows/apisix-docker-example-test.yaml [hunk 2]
```diff
@@ -96,4 +78,4 @@ jobs:
           if [[ $result_code -ne 200 ]];then
                   printf "result_code: %s\n" "$result_code"
                   exit 125
-          fi
+          fi
```
Comment (tokers) [inline .github/workflows/apisix-docker-example-test.yaml:81]: add EOL.

**Review by soulbird — APPROVED** [review]

File: .github/workflows/apisix-docker-example-test.yaml (hunk 2 above)
Comment (AlinsRan) [reply .github/workflows/apisix-docker-example-test.yaml]: There is EOL, but I don't seem to see it?

**Pull request merged**